### PR TITLE
Re-enable OSMesa builds and add EGL build variant

### PR DESCRIPTION
**Pull request by basnijholt**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This closes the issue https://github.com/conda-forge/vtk-feedstock/issues/223, and superseded  https://github.com/conda-forge/vtk-feedstock/pull/153 and https://github.com/conda-forge/vtk-feedstock/pull/224.

I have locally built the
- EGL variant on Linux for Python 3.9, 3.10
- OSMesa variant on Linux for Python 3.9, 3.10
- qt variant on Linux for Python 3.9, 3.10

We're not using the external version of GLEW for EGL/OSMesa, because GLEW needs to be compiled with EGL/OSMesa.

### CODE REVIEW

**Comment by basnijholt** [comment]
Currently, *all* pipelines succeed in compiling, except the `linux_64_build_variantosmesapython3.10` workflow. I have no idea why that specific one doesn't work.

The tests still fail because it seems like we cannot yum install the `mesa-libGL-devel` lib.

**Comment by basnijholt** [comment]
@msarahan, @ccordoba12, @grlee77, @jasonb5, or @matthiasdiener, could you please merge/review this? 😄

All CI builds are passing (with the exception of one of the Windows build timing out for unrelated reasons.)

**Review by matthiasdiener — APPROVED** [review]
This looks great, thank you for your hard work @basnijholt !

**Comment by basnijholt** [comment]
@matthiasdiener, thanks!

Do we need to wait for more reviews before merging?

I am kind of blocked by not having this, so my apologies for being impatient 😄

**Comment by basnijholt** [comment]
@conda-forge-admin, please restart ci

**Comment by matthiasdiener** [comment]
By the way, would you like to add yourself as a maintainer @basnijholt ?

**Comment by basnijholt** [comment]
@matthiasdiener, sure! I have just done that.

**Comment by matthiasdiener** [comment]
@conda-forge-admin rerender

**Comment by basnijholt** [comment]
Everything is passing! ☑ :tada:

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {% set version = "9.1.0" %}
-{% set build = 2 %}
+{% set build = 3 %}
 
 {% set minor_version = ".".join(version.split(".")[:2]) %}
 
-{% set build = build + 100 %}   # [not VTK_WITH_OSMESA]
+{% set build = build + 100 %}   # [build_variant == "osmesa"]
+{% set build = build + 200 %}   # [build_variant == "qt"]
```
Comment (grlee77) [inline recipe/meta.yaml:7]: isn't this offset inconsistent with the previous setting? i.e. it was formerly +100 when NOT osmesa?

Comment (grlee77) [reply recipe/meta.yaml:7]: Or does it not matter as long as the numbers are distinct for the variants?

Comment (basnijholt) [reply recipe/meta.yaml:7]: I don’t think this matters, as long as the qt version has the highest build number such that it is preferred over the other build variants.

**Comment by grlee77** [comment]
Okay, let's try it out. Thanks, @basnijholt!

**Pull request merged**